lib/updatehub/updatehub.c

```diff
@@ -106,14 +104,8 @@ static int metadata_hash_get(char *metadata)
 		return -1;
 	}
 
-	memset(update_info.package_uid, 0, TC_SHA256_BLOCK_SIZE + 1);
-	for (int i = 0; i < TC_SHA256_DIGEST_SIZE; i++) {
-		snprintk(buffer, sizeof(buffer), "%02x",
-			 hash[i]);
-		buffer_len = buffer_len + strlen(buffer);
-		strncat(&update_info.package_uid[i], buffer,
-			MIN(TC_SHA256_BLOCK_SIZE, buffer_len));
-	}
+	bin2hex(hash, TC_SHA256_DIGEST_SIZE,
+		update_info.package_uid, TC_SHA256_BLOCK_SIZE);
```

Comment:
Again, TC_SHA256_BLOCK_SIZE has nothing to do with this value. You want something (that you define) equal to 2 * TC_SHA256_DIGEST_SIZE.

Reply:
Fixed like others.